battlepass/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+	/// ActiveBattlepassByOrg: map Hash => Hash
+	#[pallet::storage]
+	#[pallet::getter(fn get_active_battlepass)]
+	pub type ActiveBattlepassByOrg<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, T::Hash, OptionQuery>;
+
+	/// Claimed Battlepass-NFT by user account.
+	///
+	/// ClaimedBattlepass: map (AccountId, Hash) => BattlepassNft
+	#[pallet::storage]
+	#[pallet::getter(fn get_claimed_battlepass)]
+	pub(super) type ClaimedBattlepass<T: Config> = StorageDoubleMap<_,
+		Blake2_128Concat, T::AccountId,
+		Blake2_128Concat, T::Hash,
+		BattlepassNft,
+		OptionQuery
+	>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(0)]
+		pub fn create_battlepass(
+			origin: OriginFor<T>,
+			org_id: T::Hash,
+			name: String<T>,
+			cid: String<T>,
+			price: u16,
+		) -> DispatchResult {
+			let creator = ensure_signed(origin)?;
+			// check if Org exists
+			ensure!(T::Control::is_org_active(&org_id), Error::<T>::OrgUnknown);
+			// check if origin is an Org Prime
+			let prime = T::Control::org_prime_account(&org_id).ok_or(Error::<T>::OrgPrimeUnknown)?;
+			ensure!(creator == prime, Error::<T>::AuthorizationError);  												// TODO: Add Root temporaryly
+			// check is there is no active battlepass for the Org
+			ensure!(!ActiveBattlepassByOrg::<T>::contains_key(&org_id), Error::<T>::BattlepassExists);
+			let new_season = Self::get_battlepass_count(org_id) + 1;
+			let now = <frame_system::Pallet<T>>::block_number();
+
+			// Create collection 
+			let metadata = BoundedVec::truncate_from(b"meta".to_vec());		// TODO: what should be here?
+			let symbol = BoundedVec::truncate_from(b"symbol".to_vec());		// TODO: what should be here?
+			let collection_id = T::Rmrk::collection_create(creator.clone(), metadata, None, symbol)?;
+
+			// Create Battlepass
+			let battlepass: Battlepass<T::Hash, T::AccountId, T::BlockNumber, String<T>> = types::Battlepass {
+				creator, 
+				org_id,
+				name,
+				cid,
+				season: new_season,
+				collection_id,
+				price,
+				created: now.clone(), mutated: now
+			};
+			let battlepass_id = <T as frame_system::Config>::Hashing::hash_of(&battlepass);
+
+			Battlepasses::<T>::insert(&battlepass_id, battlepass);
+			BattlepassStates::<T>::insert(&battlepass_id, types::BattlepassState::Active); 								// TODO: change to Draft
+			ActiveBattlepassByOrg::<T>::insert(org_id, battlepass_id);
+			BattlepassCount::<T>::insert(org_id, new_season);
+
+			Self::deposit_event(Event::BattlepassCreated { org_id, battlepass_id, season: new_season });
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn claim_battlepass(
+			origin: OriginFor<T>,
+			org_id: T::Hash,
+		) -> DispatchResult {
+			let claimer = ensure_signed(origin)?;
+			// check if Org exists
+			ensure!(T::Control::is_org_active(&org_id), Error::<T>::OrgUnknown);
+			// check if user is a member of organization
+			ensure!(T::Control::is_org_member_active(&org_id, &claimer), Error::<T>::NotMember);
+			// check if Org has active battlepass
```

Comment:
if you look to the two checks above maybe we should introduce the same for battlepass, like `T::Battlepass::is_battlepass_active` ?

Reply:
Here we need to get the active battlepass, not just check if it's active.